### PR TITLE
[UES-898] Making https default for login operation

### PR DESCRIPTION
**Pull request by Megala21**
This will redirect the http request coming from the users to login to https request to ensure it is served through secure channel.

Public Jira -[https://wso2.org/jira/browse/UES-898]

### CODE REVIEW

File: apps/portal/routers/pages.jag [hunk 1]
```diff
@@ -69,6 +69,20 @@ var isMobileDevice = false;
         response.sendError(404, 'Requested page cannot be found');
         return;
     }
+
+    // Check whether a request for login and whether it is from secured channel
+    if(controller.localeCompare("/controllers/login-controller.jag") == 0 && !request.isSecure()){
+        var proc = require('process');
+
+        // If it is not secure redirect it to a secure channel
+        var newURL = request.getRequestURL().replace(request.getLocalPort(), proc.getProperty('mgt.transport.https.port'));
```
Comment (sinthuja) [inline apps/portal/routers/pages.jag]: Please do validate the above fix works when server starts with 443, and 80 port respectively.

Comment (Megala21) [inline apps/portal/routers/pages.jag]: I have tested it and it works well for those ports as well

File: apps/portal/routers/pages.jag [hunk 2]
```diff
@@ -69,6 +69,20 @@ var isMobileDevice = false;
         response.sendError(404, 'Requested page cannot be found');
         return;
     }
+
+    // Check whether a request for login and whether it is from secured channel
+    if (controller.localeCompare("/controllers/login-controller.jag") == 0 && !request.isSecure()) {
+        var proc = require('process');
+
```
Comment (DilanUA) [inline apps/portal/routers/pages.jag]: unnecessary line

File: apps/portal/routers/pages.jag [hunk 3]
```diff
@@ -69,6 +69,20 @@ var isMobileDevice = false;
...
+        // If it is not secure redirect it to a secure channel
+        var newURL = request.getRequestURL().replace(request.getLocalPort(), proc.getProperty('mgt.transport.https.port'));
```
Comment (DilanUA) [inline apps/portal/routers/pages.jag]: better be consistent with quotations for defining strings, use double quotes for outer strings and single quotes for inner strings.

File: apps/portal/routers/pages.jag (hunk 2 above)
Comment (Megala21) [inline apps/portal/routers/pages.jag]: Fixed with 7bc4c4eb04408d0696649f4c7dee8bfdda80f364

File: apps/portal/routers/pages.jag (hunk 3 above)
Comment (Megala21) [inline apps/portal/routers/pages.jag]: Fixed with 7bc4c4eb04408d0696649f4c7dee8bfdda80f364

**Pull request merged**